src/api_helpers.jl

```diff
@@ -82,6 +82,24 @@ function h5d_vlen_get_buf_size(dataset_id, type_id, space_id)
     return sz[]
 end
 
+function h5d_get_chunk_info(dataset_id, fspace_id, index)
+    offset = Vector{hsize_t}(undef, ndims(dataset_id))
+    filter_mask = Ref{UInt32}(0)
+    addr = Ref{HDF5.haddr_t}(0)
+    size = Ref{HDF5.hsize_t}(0)
+    h5d_get_chunk_info(dataset_id, fspace_id, index, offset, filter_mask, addr, size)
+    return (offset = offset, filter_mask = filter_mask[], addr = addr[], size = size[])
```

Comment:
@jmert  We haven't been using named tuples for the returns, I'm not sure if we should remove it here or not, thoughts?

Reply:
I considered three options
1. `Tuple`
2. `NamedTuple`
3. `struct`

`Tuple` is difficult because there are so many parameters being returned, but this would be consistent with the rest of the API.

`NamedTuple` is compatible with `Tuple` in that you can also index this numerically. Accepting this would suggest that we should make all the other midlevel helpers `NamedTuple` as well though.

`struct` just creates extra type noise and not quite compatible with the previous API-style without additional effort.

Ultimately, I think `NamedTuple` is a decent compromise. If you really wanted just a `Tuple`, `value(::NamedTuple)` allows you to extract it easily. While this suggest changing other elements to a `NamedTuple`, that does not necessarily need to be case if it is too cumbersome.

Reply:
> @jmert We haven't been using named tuples for the returns, I'm not sure if we should remove it here or not, thoughts? I think it is probably a good idea, albeit this is the mid-layer interface. There two other methods in the `api_helpers.jl` file that could use a named-tuple output return type ...

I think it'd be fine to use named tuples in the existing functions — as long as the order is the same as the tuple now, destructuring assignment will still iterate correctly and should be fine in terms of backwards compatibility.

Reply:
Sounds good, let's stick to the `NamedTuple` option then. Might want to rename `addr` to `address`? I don't mind either, but it might be more 'julian' to spell it out.

Reply:
Great. I'll work on this, and also see if I can turn my demo into a test using one of the existing test datasets. Thanks for the feedback.

Reply:
Albeit leaving it as `addr` is sticking to the lib-hdf5 convention and as this is the mid-level api, leaving it as is might be better.